server/src/main/java/io/crate/types/NumericType.java

```diff
@@ -158,6 +159,10 @@ public BigDecimal sanitizeValue(Object value) {
             BigInteger bigInt = BigInteger.valueOf(longValue);
             return new BigDecimal(bigInt, scale == null ? 0 : scale, mathContext());
         }
+        if (value instanceof byte[] bytes) {
+            var bigInt = NumericUtils.sortableBytesToBigInt(bytes, 0, bytes.length);
+            return new BigDecimal(bigInt, scale == null ? 0 : scale, mathContext());
+        }
```

Comment:
Doesn't the format depend on the precision? See `NumericStorage.getCollectorExpression`